test/test_port.rb

```diff
@@ -363,6 +363,110 @@ def configure
             port_reader.reader.port
     end
 
+    describe "#read_new" do
+        before do
+            @task = syskit_stub_deploy_and_configure(task_m, remote_task: false)
+            @port_reader = @task.out_port.reader
+            @orocos_port = Orocos.allow_blocking_calls do
+                @task.orocos_task.raw_port('out')
+            end
+        end
+        it "returns nil if the reader is not yet connected" do
+            refute @port_reader.connected?
+            assert_nil @port_reader.read_new
+        end
+        describe "without an explicitely-provided sample" do
+            before do
+                syskit_wait_ready(@port_reader)
+            end
+            it "reads new samples" do
+                @orocos_port.write(10)
```

Comment:
if I run:
syskit test test_port.rb
I get:
NoMethodError: undefined method `write' for stub67.out:Orocos::OutputPort

Reply:
This test requires https://github.com/rock-core/tools-orocosrb/pull/123

Reply:
Also, within the Ruby packages (i.e. outside of a bundle), just use `ruby`

~~~
ruby -Ilib -Itest test/test_port.rb
~~~

Reply:
Elael's just reviewed the orocos.rb PR. Could you update it and check if the tests pass again ?

Reply:
ok! everything is working now.